src/modules/party/Category.ts

```diff
@@ -33,55 +33,74 @@ export default class PokemonCategories implements Saveable {
                 p.category = 0;
             }
         });
-        let categoryIndex = this.categories().length;
-        while (categoryIndex-- > 0) {
-            this.removeCategory(categoryIndex, true);
+        let max = 0;
+        PokemonCategories.categories().forEach(c => {
+            max = Math.max(max, c.id);
+        });
+        while (max >= 0) {
+            PokemonCategories.removeCategory(max, true);
+            max--;
         }
```

Comment:
This is kind of odd, why don't we just do:
```suggestion
        PokemonCategories.categories().forEach(
            c => PokemonCategories.removeCategory(c.id, true),
        );
```

Reply:
Sure, I can optimise that code. I do not know why I stuck with that.
However, `forEach` does not work because of the use of `splice`.

Reply:
you can take a copy with `[...PokemonCategories.categories()]` and forEach over that

Reply:
Done. Should be fine.